### PR TITLE
physics: Unit dyads access from ReferenceFrames

### LINKED ISSUE #24965: ReferenceFrame should have attributes to access unit dyads and its unit dyadic
ReferenceFrame has `.x,.y,.z` to access the unit vectors. It would be nice if you could access unit dyads also with `.xx,.xy,.xz,...`. Then you can construct dyadics without having to do `outer(N.x, N.x)`. This is a simple fix and addition. It would need to be documented and shown in some examples.

### PR DESCRIPTION
**Pull request by kostas-rigan**
#### References to other Issues or PRs

fixes #24965

#### Brief description of what is fixed or changed
Unit dyads and unit dyadic can be accessed through attributes of a ReferenceFrame object. It is an addition related to issue #24965, allowing easy access to unit dyads without the need to use the outer() function. This commit adds 10 functions with the property decorator, along with a few examples in the constructor's documentation.

#### Release Notes

* physics.vector
    * Added access to unit dyads and unit dyadic directly from ReferenceFrame objects.

### CODE REVIEW

**Comment by moorepants** [comment]
This is looking good so far. You have some trailing whitespaces that the linter doesn't like:

```
sympy/testing/tests/test_code_quality.py:236: in test_this_file
    assert False, message_space % (fname, idx + 1)
E   AssertionError: File contains trailing whitespace: /home/runner/work/sympy/sympy/sympy/physics/vector/frame.py, line 1403.
E   assert False
```

**Comment by moorepants** [comment]
The only think this needs are some simple unit tests to check the new functionality. Can you add a basic test for each of the new attributes?

**Comment by kostas-rigan** [comment]
Yes I will write some tests. Sorry, I forgot to run the tests previously and didn't see the trailing whitespaces. 
I've also worked on adding the unit dyadic, and it looks like this:
``` python
@property
def dyadic(self):
    """Unit dyadic for the ReferenceFrame."""
    return Matrix([[1, 0, 0],
                   [0, 1, 0],
                   [0, 0, 1]])
```
I will write a test for it as well and then push everything together.

**Comment by moorepants** [comment]
The unit dyadic should return a dyadic not a matrix.

It needs a better name too. We can do `ReferenceFrame.unit_dyadic` as that is the clearest. I was trying to think about a simpler name though because you may want to use it in math equations. Take the example here https://moorepants.github.io/learn-multibody-dynamics/mass.html#parallel-axis-theorem

```python
dx, dy, dz, m = sm.symbols('d_x, d_y, d_z, m')

N = me.ReferenceFrame('N')

r_O_Bo = dx*N.x + dy*N.y + dz*N.z

U = me.outer(N.x, N.x) + me.outer(N.y, N.y) + me.outer(N.z, N.z)

I_Bo_O = m*(me.dot(r_O_Bo, r_O_Bo)*U - me.outer(r_O_Bo, r_O_Bo))
```

it could be written like:

```python
I_Bo_O = m*(me.dot(r_O_Bo, r_O_Bo)*N.u - me.outer(r_O_Bo, r_O_Bo))
```

or something. Much like our `N.x,N.xx,`.

**Comment by moorepants** [comment]
I guess `N.unit` could work. It is shorter and maybe descriptive enough. I'll think about the best name a bit.

@TJStienstra any preferences on a name for a unit dyadic?

**Comment by abhishek-kuma** [comment]
This is my first time contributing to an open source, so I'm less experienced. The issue is almost solved. I'm actually contributing for GSoC - 2023 . I think i have to learn how to add tests now .

**Comment by moorepants** [comment]
>  I think i have to learn how to add tests now .

Yes, all PRs should have some unit tests.

**Comment by kostas-rigan** [comment]
> The unit dyadic should return a dyadic not a matrix.
> 
I ran your example, and I changed the code to:
``` python
@property
def unit(self):
    """Unit dyadic for the ReferenceFrame."""
    return self.xx + self.yy + self.zz
```

**Comment by kostas-rigan** [comment]
@moorepants is it preferable to squash the commits into a single one? Due to some mistakes I made, there are a couple of commits for minor changes that I could have avoided.

**Comment by moorepants** [comment]
> @moorepants is it preferable to squash the commits into a single one? Due to some mistakes I made, there are a couple of commits for minor changes that I could have avoided.

Squashing is not necessary or very common in SymPy commit history, but if you know what you are doing and want to force push a cleaned up commit history you can.

File: sympy/physics/vector/frame.py [hunk 1]
```diff
@@ -145,6 +145,30 @@ def __init__(self, name, indices=None, latexs=None, variables=None):
...
+        >>> P.xx
+        (P['1']|P['1'])
+        >>> P.zy
+        (P['3']|P['2'])
+
+        Unit dyadic is also accessible via the ``u`` attribute:
+
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> N = ReferenceFrame('N')
+        >>> N.unit
```
Comment (moorepants) [inline sympy/physics/vector/frame.py:166]:
```suggestion
        >>> N.u
```

**Comment by moorepants** [comment]
If you fix the minor error in the doctest and then update the release notes to reflect adding unit dyads and the unit dyadic, then this looks good to me.

**Comment by kostas-rigan** [comment]
> If you fix the minor error in the doctest and then update the release notes to reflect adding unit dyads and the unit dyadic, then this looks good to me.

Great, I will fix that right away. Thank you for the guidance!

**Comment by kostas-rigan** [comment]
> #### References to other Issues or PRs
> fixes #24965
> 
> #### Brief description of what is fixed or changed
> Unit dyads and unit dyadic can be accessed through attributes of a ReferenceFrame object. It is an addition related to issue #24965, allowing easy access to unit dyads without the need to use the outer() function. This commit adds 10 functions with the property decorator, along with a few of examples in the constructor documentation.
> 
> #### Other comments
> 
> 
> #### Release Notes
> * physics.vector
>   * Added access to unit dyads and unit dyadic directly from ReferenceFrame objects.

**Comment by moorepants** [comment]
Thanks for contribution. If tests pass it will merge!

**Pull request merged**